src/js/background/background.js

```diff
@@ -87,10 +87,10 @@ async function patchMaskInfo(method = "PATCH", id, data, opts=null) {
   return await response.json();
 }
 
-async function storePremiumAvailabilityInCountry() {
+async function storeRuntimeData() {
   // If we already fetched Premium availability in the past seven days,
   // don't fetch it again.
-  const existingPremiumAvailability = (await browser.storage.local.get("premiumCountries"))?.premiumCountries;
+  const existingPremiumAvailability = (await browser.storage.local.get("premiumCountries")).premiumCountries?.PREMIUM_PLANS;
   if (typeof existingPremiumAvailability === "object" && existingPremiumAvailability.fetchedAt > (Date.now() - 7 * 24 * 60 * 60 * 1000)) {
     return;
   }
```

Comment:
Hmm yeah, I'd expect that to happen. If the `introPricingEndDate` is not found, [`new Date(undefined)` is called](https://github.com/mozilla/fx-private-relay-add-on/blob/49aab832bf386a754ee25da0c2c14a62ee202e59/src/js/popup/popup.js#L441-L442), which results in an invalid Date object, for which `.getTime()` returns `NaN`, which I'd expect to cause all kinds of problems.

I think to be sure we'd probably best disregard whatever value is already stored for `premiumCountries` in local storage (possibly even just deleting any existing values, if present), and then instead checking for a new value `runtimeData` instead?

Reply:
> Edit: though of course, that might be problematic for other references to `premiumCountries` in the code. Ugh, caching and no types... OK, then alternatively instead of just checking for `existingPremiumAvailability.fetchedAt`, we check whether either that _or_ existingIntroPricingEndDate.fetchedAt` is nonexistent or more than seven days ago?

Yeah, I can't believe I find myself WANTING types now. 😜 

I agree - we should probably check for both `existingPremiumAvailability` AND ` existingIntroPricingEndDate` - and if either of them doesn't exist or is 7+ day sold, re-fetch.